### PR TITLE
Added autoClose behaviour for dropdowns

### PR DESCRIPTION
**Pull request by jotoh98**
More about the autoClose behaviour [here](https://getbootstrap.com/docs/5.0/components/dropdowns/#auto-close-behavior).

Additionally to the implementation, I've added

- a playground example
- a documentation section
- unit-tests

Preview: https://deploy-preview-5873--react-bootstrap.netlify.app/

### CODE REVIEW

File: src/Dropdown.tsx [hunk 1]
```diff
@@ -156,8 +165,20 @@ const Dropdown: BsPrefixRefForwardingComponent<'div', DropdownProps> =
         if (
           event.currentTarget === document &&
           (source !== 'keydown' || event.key === 'Escape')
-        )
+        ) {
           source = 'rootClose';
+
+          const noOuterCloseModes =
```
Comment (kyletsang) [inline src/Dropdown.tsx:171]: Might be worth leaving comments about click what elements triggers the closing of the dropdown so it's more clear

File: test/DropdownSpec.js [hunk 2]
```diff
@@ -333,4 +333,141 @@ describe('<Dropdown>', () => {
       ).assertSingle('div.dropdown-menu');
     });
   });
+
+  describe('autoClose behaviour', () => {
+    const createDocumentListenersMock = () => {
```
Comment (kyletsang) [inline test/DropdownSpec.js:338]: Do we need this?  We have other methods in other tests to trigger click events.

**Review by kyletsang — COMMENTED** [review]
Overall, looks good.  Thanks for taking this on.  Left a few comments.

File: src/Dropdown.tsx [hunk 3]
```diff
@@ -47,6 +47,7 @@ export interface DropdownProps
   focusFirstItemOnShow?: boolean | 'keyboard';
   onSelect?: SelectCallback;
   navbar?: boolean;
+  autoClose: 'true' | 'outside' | 'inside' | 'false';
```
[the remark this answers is not in the document]
Comment (jotoh98) [reply src/Dropdown.tsx:50]: Yes, I thought of that as well. Wasn't sure about the consistency but I will change that.

File: test/DropdownSpec.js (hunk 2 above)
Comment (jotoh98) [reply test/DropdownSpec.js:338]: I didn't notice any method concerned with simulating clicks outside of the component. Correct me if I'm wrong. Should we keep the outside click methods?

Comment (kyletsang) [reply test/DropdownSpec.js:338]: We did use `simulant.fire(document.body, 'click');` inside an Offcanvas test.  Can we try that and see if it works for your test?

File: www/src/examples/Dropdown/AutoClose.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<>
+  {[true, 'outside', 'inside', false].map((autoClose, index) => (
```
Comment (kyletsang) [inline www/src/examples/Dropdown/AutoClose.js:2]: Wonder if we should rearrange/reword the examples so it mirrors upstream Bootstrap examples:
https://getbootstrap.com/docs/5.0/components/dropdowns/#auto-close-behavior

I got thrown off with the autoclose behavior for "inside" and "outside" differing from Bootstrap's docs, but turns out it was an issue of ordering/wording.

**Review by kyletsang — APPROVED** [review]

**Comment by kyletsang** [comment]
Looks good to me - left 1 comment regarding docs demo.

@jquense, would be good to get your eyes on this.

**Review by jquense — APPROVED** [review]
LGTM, happy with how this was relatively straightforward to implement outside of react-overlays this is

**Comment by kyletsang** [comment]
Thanks!

**Pull request merged**